thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobSchedulerUtils.java

```diff
@@ -117,14 +118,28 @@ public static long getBucketSizeInMSForDataset(DatasetConfigDTO datasetConfig, A
     // Calculate time periods according to the function frequency
     if (dataUnit.equals(TimeUnit.MINUTES) || dataUnit.equals(TimeUnit.MILLISECONDS) || dataUnit.equals(TimeUnit.SECONDS)) {
       if (functionFrequency.getUnit().equals(TimeUnit.MINUTES) && (functionFrequency.getSize() <=30)) {
-        bucketMillis = TimeUnit.MILLISECONDS.convert(functionFrequency.getSize(), TimeUnit.MINUTES);
+        bucketSizePeriod = new Period(0, 0, 0, 0, 0, functionFrequency.getSize(), 0, 0);
       } else {
-        bucketMillis = TimeUnit.MILLISECONDS.convert(1, TimeUnit.HOURS); // default to HOURS
+        bucketSizePeriod = getBucketSizePeriodForUnit(TimeUnit.HOURS); // default to 1 HOUR
       }
     } else {
-      bucketMillis = TimeUnit.MILLISECONDS.convert(1, dataUnit);
+      bucketSizePeriod = getBucketSizePeriodForUnit(dataUnit);
     }
-    return bucketMillis;
+    return bucketSizePeriod;
+  }
+
+  private static Period getBucketSizePeriodForUnit(TimeUnit unit) {
```

Comment:
I suggest that we add a method (e.g., toPeriod) in TimeGranularity.java that returns the period according to time unit and size of the time granularity. We can also take care of lines 119-127 in toPeriod method.

We also need to consider the case of MINUTES.